### PR TITLE
wgsl: Add normalize builtin validation tests

### PR DESCRIPTION
**Pull request by toji**
Adds validation tests for `normalize()`.

Took a while to get the right logic for testing the validity of the numeric values which didn't result in false positives, but at this point all the test cases pass. More than happy to take feedback on how that's being computed, though!

Also, I have no idea what's going on with the changes to hashes.json. I'm assuming it's because I reorganized the scalar/vector types in `conversion.js` in order to get lists of *only* the vector types?

<hr>

**Requirements for PR author:**

- [ ] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [ ] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [ ] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by toji** [comment]
Ugh. Noticed that this is failing on devices with f16 support, I'll work on that.

**Comment by toji** [comment]
Okay, think I worked out all the kinks, just in time for the weekend.

File: src/webgpu/shader/validation/expression/call/builtin/normalize.spec.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  kConstantAndOverrideStages,
+  stageSupportsType,
+  validateConstOrOverrideBuiltinEval,
+} from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kValidArgumentTypes = objectsToRecord(kConvertableToFloatVectors);
+
+// These should probably go in a utilities file somewhere?
```
Comment (zoddicus) [inline src/webgpu/shader/validation/expression/call/builtin/normalize.spec.ts:27]: Looks like you are just doing fp quanitzation, so I think you should be able to use quantizeToF32 and quantizeToF16 from src/webgpu/util/math.ts

**Review by zoddicus — CHANGES_REQUESTED** [review]

File: src/webgpu/shader/validation/expression/call/builtin/normalize.spec.ts (hunk 1 above)
Comment (toji) [reply src/webgpu/shader/validation/expression/call/builtin/normalize.spec.ts]: You're right! Thanks for pointing out those functions!

**Review by zoddicus — APPROVED** [review]

**Pull request merged**